### DISCUSSION TITLE
Hide the validation messages and make the edges of the input red instead

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#10645] [Help]
How can I hide the validation messages and make the edges of the input red instead?

**Comment by leandrocfe** [comment]
1 - Create a [custom theme](https://filamentphp.com/docs/3.x/panels/themes#creating-a-custom-theme)
2 - Add in the css

```css
.fi-fo-field-wrp-error-message {
    @apply hidden;
}
```

3 - Use validation rules

```php
Forms\Components\TextInput::make('name')
    ->rule('min:5')
```

**Reply by ArtMin96** [reply]
Thanks, Leandro :+1:  I am more interested in how to make the border of the field red. I have done this.

**Reply by ArtMin96** [reply]
I looked into the component and did not find an option.

**Reply by leandrocfe** [reply]
It should be red

**Reply by ArtMin96** [reply]
Sorry, I forgot to emphasize that I used the form package in a separate project.

**Reply by ArtMin96** [reply]
Sorry, the problem was with CSS. Thanks again

**Reply by z4yed** [reply]
How do we remove the border of the TextInput field completely?

**Reply by leandrocfe** [reply]
you need to customize via css

**Reply by myregistration** [reply]
Thank you leandrocfe!  

I modified class so I can use `extraAttribute(['class' => 'hide-error-messages']` for particular fields, including a repeater.

in `app.css`

```
.hide-error-messages .fi-fo-field-wrp-error-message {
      @apply hidden;
}
```

If you want to display a notification with error messages you can use `onValidationError()`.  I'm using a Livewire Component, but maybe it's the same function in a Filament Resource page:

  ```
 protected function onValidationError(ValidationException $exception): void
    {
        $errorMessages = [];

        foreach ($exception->errors() as $field => $messages) {

            foreach ($messages as $message) {

                // Customize the message format if needed
                $errorMessages[] = "$message";
            }
        }

        Notification::make()
            ->title('Validation Failed')
            ->body(implode('<br>', $errorMessages)) // Join messages with line breaks
            ->danger()
            ->send();

        // or you can use the default message summary
        // Notification::make()
        //     ->title($exception->getMessage())
        //     ->danger()
        //     ->send();
    }
```
